From fa6e4a4cb81132721f35ea5240562e49f8f98e54 Mon Sep 17 00:00:00 2001
From: Trung Nguyen <57174311+trungnt2910@users.noreply.github.com>
Date: Fri, 31 Jul 2026 23:50:18 +1000
Subject: [PATCH] Refactor DefPointerList with compliant C++

Use qualified names when accessing template parent class members.

This allows MRTCore to be compiled with Clang on MinGW.
---
 dev/MRTCore/mrt/mrm/include/mrm/build/DefList.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dev/MRTCore/mrt/mrm/include/mrm/build/DefList.h b/dev/MRTCore/mrt/mrm/include/mrm/build/DefList.h
index 6feee0949a..6720023e31 100644
--- a/dev/MRTCore/mrt/mrm/include/mrm/build/DefList.h
+++ b/dev/MRTCore/mrt/mrm/include/mrm/build/DefList.h
@@ -560,6 +560,8 @@ class DefList : public DefObject
 template<typename VALUE, typename CallBackCompareFunc = VoidComparerFunc, typename CallbackHashFunc = VoidHashFunc>
 class DefPointerList : public DefList<VALUE*, CallBackCompareFunc, CallbackHashFunc>
 {
+private:
+    using Base = DefList<VALUE*, CallBackCompareFunc, CallbackHashFunc>;
 public:
     static HRESULT CreateInstance(_Outptr_ DefPointerList<VALUE, CallBackCompareFunc, CallbackHashFunc>** result)
     {
@@ -607,10 +609,10 @@ class DefPointerList : public DefList<VALUE*, CallBackCompareFunc, CallbackHashF
 
     virtual ~DefPointerList()
     {
-        for (int i = 0; i < DefList::Count(); i++)
+        for (int i = 0; i < Base::Count(); i++)
         {
             const VALUE* pMember = nullptr;
-            if (TryGetValue(i, &pMember))
+            if (Base::TryGetValue(i, &pMember))
             {
                 delete (pMember);
             }
